metron-analytics/metron-profiler-client/src/main/java/org/apache/metron/profiler/client/stellar/GetProfile.java

```diff
@@ -132,35 +144,33 @@
 
   private static final Logger LOG = LoggerFactory.getLogger(GetProfile.class);
 
-  /**
-   * A client that can retrieve profile values.
-   */
+
+  // Global configuration from client context, used to help create profiler client.
+  private Map<String, Object> global;
+
+  // Cached client that can retrieve profile values.
   private ProfilerClient client;
 
+  // Cached value of config override map used to construct the previously cached client.
+  private Map cachedConfigOverrideMap = null;
+
   /**
    * Initialization.
    */
   @Override
   public void initialize(Context context) {
-
     // ensure the required capabilities are defined
     Context.Capabilities[] required = { GLOBAL_CONFIG };
     validateCapabilities(context, required);
-    Map<String, Object> global = (Map<String, Object>) context.getCapability(GLOBAL_CONFIG).get();
-
-    // create the profiler client
-    RowKeyBuilder rowKeyBuilder = getRowKeyBuilder(global);
-    ColumnBuilder columnBuilder = getColumnBuilder(global);
-    HTableInterface table = getTable(global);
-    client = new HBaseProfilerClient(table, rowKeyBuilder, columnBuilder);
+    global = (Map<String, Object>) context.getCapability(GLOBAL_CONFIG).get();
```

Comment:
If the global config changes, you probably want to pick that up post-initialization.  I would calling `context.getCapability(GLOBAL_CONFIG).get()` from the apply rather than caching a copy.